<?xml version="1.0"?>
<launch>]
        <!-- Python Version: If using ROS Noetic or Python3 in other versions of ROS, set this to true. If using versions prior to ROS Noetic and ubuntu 20.04, mark as false-->
        <arg name="python3" default="true"/>
        	
	<!-- usb_cam -->
        <arg name="run_usb_cam" default="true"/>
        
        <!-- MediaPipe ROS -->
        <param name="/mediapipe_holistic_ros/input_usb_cam_topic" value="/usb_cam/image_raw" />
        <param name="/mediapipe_holistic_ros/output_image_topic" value="/mediapipe_holistic/output_image" />
        <param name="/mediapipe_holistic_ros/output_mediapipe_topic" value="/MediaPipePose/holistic/landmarks" />
        <param name="/mediapipe_holistic_ros/pub_landmark_output" value="true" />
        <param name="/mediapipe_holistic_ros/pub_image_output" value="true" />
        <param name="/mediapipe_holistic_ros/open_view_image" value="true" />
      
        
	<group if="$(arg run_usb_cam)">
                <node name="usb_cam" pkg="usb_cam" type="usb_cam_node" output="screen" >
                    <param name="video_device" value="/dev/video0" />
                    <param name="image_width" value="640" />
                    <param name="image_height" value="480" />
                    <param name="pixel_format" value="yuyv" />
                    <param name="camera_frame_id" value="usb_cam" />
                    <param name="io_method" value="mmap"/>
                </node>
        </group>        

        <group if="$(arg python3)">
                <node pkg="mediapipe_holistic_ros" type="mediapipe_holistic_python3.py" name="mediapipe_holistic_python3" output="screen" />
        </group> 
        
</launch>
